module/extension/Director/src/Director.hpp

```diff
@@ -62,6 +62,22 @@ namespace module::extension {
          */
         void add_causing(const ::extension::behaviour::commands::CausingExpression& causing);
 
+        /**
+         * Compares the priorities of two director tasks and returns true if the challenger has priority over the
+         * incumbent.
+         *
+         * The function requires that the challengers precendence is strictly greater than the incumbent.
+         * This ensures that we don't change tasks unnecessarily when the priority is equal.
+         *
+         * @param incumbent     the task to compare which is currently the active running task
+         * @param challenger    the task to compare which wants to run but is not currently
+         *
+         * @return true     if the challenger has strictly higher priority than the incumbent
+         * @return false    if the incumbent task has equal or higher priority
+         */
+        bool compare_priority(const std::shared_ptr<const ::extension::behaviour::commands::DirectorTask>& incumbent,
```

Comment:
I would also consider changing this variable name. This is not a commutative operation, but the name implies it could be? I could also be reading too far into this

Reply:
which variable are you talking about?

Reply:
Open to name suggestions for this function, I've changed it like 3 times before it even made it to a PR

Reply:
I was indeed talking about the function, wrong word. `challenge_priority` perhaps?

Reply:
Sure, unique words make it easier to understand concepts

Reply:
I think `compare_priority` is right. Comparisons dont have to be commutative (strictly speaking), but I think it might be in this case?